reo/validators.py

```diff
@@ -2076,3 +2020,34 @@ def chp_checks(self, params, object_name_path, number):
         else:
             self.update_attribute_value(object_name_path, number, 'tech_size_for_cost_curve', [])
 
+    def validate_chp_unavailability_periods(year, chp_unavailability_periods):
+        """
+        Validate chp_unavailability_periods and return the list of errors to append to self.input_data_errors
+        Returning a list of errors instead of directly appending to self.input_data_errors so we can reuse in views.py
+        """
+        chp_unavailability_periods_input_data_errors = []
+        valid_keys = ['month', 'start_week_of_month', 'start_day_of_week', 'start_hour', 'duration_hours']
+        for period in range(len(chp_unavailability_periods)):
+            if isinstance(chp_unavailability_periods[period], dict):
+                for key, value in chp_unavailability_periods[period].items():
```

Comment:
I think you need one more check to make sure chp_unavailability_periods[period] isn't missing any keys?

Reply:
Good catch! Done.